app/javascript/mastodon/features/compose/components/character_counter.js

```diff
@@ -1,25 +1,53 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { defineMessages, injectIntl } from 'react-intl';
 import { length } from 'stringz';
 
-export default class CharacterCounter extends React.PureComponent {
+const messages = defineMessages({
+  remaining: { id: 'character_counter.remaining', defaultMessage: 'character remaining' },
+  remaining_plural: { id: 'character_counter.remaining_plural', defaultMessage: 'characters remaining' },
+  exceeded: { id: 'character_counter.exceeded', defaultMessage: 'You have exceeded the character limit by' },
```

Comment:
The translation strings should include a placeholder for the characters count, as the word order depends on language. Furthermore, the handling of plurals also depend on the language, but that's handled by `react-intl`.

You should have something like:
```suggestion
  remaining: { id: 'character_counter.remaining', defaultMessage: '{count, plural, one {{count} character} other {{count} characters}} remaining' },
  exceeded: { id: 'character_counter.exceeded', defaultMessage: 'You have exceeded the character limit by {count}' },
```